core/src/mender-client.c

```diff
@@ -585,10 +585,8 @@ mender_client_exit(void) {
     mender_scheduler_mutex_give(mender_client_network_mutex);
     mender_scheduler_mutex_delete(mender_client_network_mutex);
     mender_client_network_mutex = NULL;
-    if (NULL != mender_client_deployment_data) {
-        cJSON_Delete(mender_client_deployment_data);
-        mender_client_deployment_data = NULL;
-    }
+    cJSON_Delete(mender_client_deployment_data);
+    mender_client_deployment_data = NULL;
```

Comment:
Time for `FREE_AND_NULL(what, destructor)`?

Reply:
I would expect `FREE_AND_NULL()` to call `free()`. Should there be two macros? E.g. `FREE_AND_NULL(what)` and `DESTROY_AND_NULL(destructor, what)`